static/app/components/devtoolbar/components/replay/replayPanel.tsx

```diff
@@ -0,0 +1,126 @@
+import {Fragment, useEffect, useState} from 'react';
+import {css} from '@emotion/react';
+
+import {Button} from 'sentry/components/button';
+import SentryAppLink from 'sentry/components/devtoolbar/components/sentryAppLink';
+import {resetFlexRowCss} from 'sentry/components/devtoolbar/styles/reset';
+import ProjectBadge from 'sentry/components/idBadge/projectBadge';
+import {IconPause, IconPlay} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import type {PlatformKey} from 'sentry/types/project';
+
+import useConfiguration from '../../hooks/useConfiguration';
+import {panelInsetContentCss, panelSectionCss} from '../../styles/panel';
+import {smallCss} from '../../styles/typography';
+import PanelLayout from '../panelLayout';
+
+const TRUNC_ID_LENGTH = 16;
+const POLL_INTERVAL_MS = 3000;
+
+export default function ReplayPanel() {
+  const {projectSlug, projectId, projectPlatform, trackAnalytics, SentrySDK} =
+    useConfiguration();
+  const replay =
+    SentrySDK && 'getReplay' in SentrySDK ? SentrySDK.getReplay() : undefined;
+
+  const [replayId, setReplayId] = useState<string | undefined>(() =>
+    replay?.getReplayId()
+  );
+  // Polls periodically since a replay could be started by sessionSampleRate
+  useEffect(() => {
+    const intervalId = setInterval(
+      () => setReplayId(replay?.getReplayId()),
+      POLL_INTERVAL_MS
+    );
+    return () => clearInterval(intervalId);
```

Comment:
```suggestion
    return clearInterval(intervalId);
```
same idea

Reply:
useEffect can optionally return a cleanup function

Reply:
the useEffect cleanup is needed here. `return () => clearInterval(interfaceId);` is the cleanup. 👍